portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2771,6 +2777,80 @@ private long _getValidFileEntryTypeId(
 		}
 	}
 
+	private void _inheritRolesPermissions(
+			long inheritableParentFolderId, DLFileEntry dlFileEntry)
+		throws PortalException {
+
+		long companyId = dlFileEntry.getCompanyId();
+		String parentClassName = DLFolderConstants.getClassName();
+
+		if (inheritableParentFolderId ==
+				DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
+
+			int count =
+				_resourcePermissionLocalService.getResourcePermissionsCount(
+					companyId, DLConstants.RESOURCE_NAME,
+					ResourceConstants.SCOPE_INDIVIDUAL,
+					String.valueOf(dlFileEntry.getGroupId()));
+
+			if (count == 0) {
+				_resourceLocalService.addResources(
+					companyId, dlFileEntry.getGroupId(), 0,
+					DLConstants.RESOURCE_NAME,
+					String.valueOf(dlFileEntry.getGroupId()), false, true,
+					true);
+			}
+
+			parentClassName = DLConstants.RESOURCE_NAME;
+			inheritableParentFolderId = dlFileEntry.getGroupId();
+		}
+
+		Map<Long, Set<String>> dlFolderRoleIdsToActionIds =
+			_resourcePermissionLocalService.
+				getAvailableResourcePermissionActionIds(
+					companyId, parentClassName,
+					ResourceConstants.SCOPE_INDIVIDUAL,
+					String.valueOf(inheritableParentFolderId),
+					ResourceActionsUtil.getModelResourceActions(
+						DLFolderConstants.getClassName()));
+
+		Map<Long, Set<String>> dlFileEntryRoleIdsToActionIds =
+			_resourcePermissionLocalService.
+				getAvailableResourcePermissionActionIds(
+					companyId, DLFileEntryConstants.getClassName(),
+					ResourceConstants.SCOPE_INDIVIDUAL,
+					String.valueOf(dlFileEntry.getFileEntryId()),
+					ResourceActionsUtil.getModelResourceActions(
+						DLFileEntryConstants.getClassName()));
+
+		Set<Long> dlFolderRoleIds = dlFolderRoleIdsToActionIds.keySet();
+
+		for (Long dlFolderRoleId : dlFolderRoleIds) {
+			Set<String> dlFolderRoleIdToActionIds =
+				dlFolderRoleIdsToActionIds.get(dlFolderRoleId);
+
+			dlFolderRoleIdToActionIds.retainAll(_commonPermissions);
+
+			Set<String> dlFileEntryActionIds = new HashSet<>(
+				dlFolderRoleIdToActionIds);
+
+			if (dlFileEntryRoleIdsToActionIds.containsKey(dlFolderRoleId)) {
+				Set<String> dlFileEntryRoleIdToActionIds =
+					dlFileEntryRoleIdsToActionIds.get(dlFolderRoleId);
+
+				dlFileEntryRoleIdToActionIds.removeAll(_commonPermissions);
+
+				dlFileEntryActionIds.addAll(dlFileEntryRoleIdToActionIds);
+			}
```

Comment:
The code could be refactored as:

```
	Set<String> dlFileEntryRoleIdToActionIds =
		dlFileEntryRoleIdsToActionIds.get(dlFolderRoleId);
		
	if (dlFileEntryRoleIdToActionIds == null) {
		continue;
	}

	dlFileEntryRoleIdToActionIds.removeAll(_commonPermissions);

	dlFileEntryActionIds.addAll(dlFileEntryRoleIdToActionIds);
```
It avoids checking if the dlFileEntryRoleIdsToActionIds map contains the dlFolderRoleId key. If the key doesn't exist, the get() method will return null.

Reply:
I think it could not be refactor as your suggestion. 
Because in the case DLFolder has Role having common permission and DLFileEntry does not have this Role, the cotinue will skip the action adding resource permission of this Role to DLFileEntry.

Reply:
Sorry, it's my fault. The code should be:
```
	Set<String> dlFileEntryRoleIdToActionIds =
		dlFileEntryRoleIdsToActionIds.get(dlFolderRoleId);
		
	if (dlFileEntryRoleIdToActionIds != null) {

		dlFileEntryRoleIdToActionIds.removeAll(_commonPermissions);

		dlFileEntryActionIds.addAll(dlFileEntryRoleIdToActionIds);
	}
```

Reply:
Fixed it. Thanks